internal/test/e2e/kubectl_test.go

```diff
@@ -341,7 +342,15 @@ func eventuallyInCluster(kubectl *kubectlContext, name string, deadline time.Dur
 	if err != nil {
 		return "", err
 	}
-
+	defer func() {
+		err := kubectl.Delete("--filename", f.Name())
+		if err != nil {
+			reterr = kerrors.NewAggregate([]error{
+				reterr,
+				fmt.Errorf("error while deleting job in eventuallyInCluster: %s ", err),
+			})
+		}
+	}()
```

Comment:
This cleans up the job once it's completed or failed....it may not be necessary, but I thought it was causing me to run out of resources in the test (see above).